### ISSUE TITLE
Security Isssue: Unauthenticated user with physical hardware access can start any app in the drawer without logging in.

### ISSUE THREAD

**Issue by stuarttempleton** [ubports/ubuntu-touch#1500]
- Device: PinePhone Ubuntu Touch CE
- Channel: Developer
- Build: 59

### Steps to reproduce

1. Make sure you have applications on the Ubuntu side drawer.
2. Lock phone
3. Wake phone up (but don't swipe in)
4. Swipe right to expose Ubuntu Touch drawer.
5. Tap on any app
6. When prompted for PIN/Password, tap cancel.
7. Repeat with a few more applications.
8. Swipe left to get PIN/Password prompt and log in.
9. Note that those apps that you started are all already running.

### Expected behavior

The selected app from the drawer only starts running if a correct login has occurred.

### Actual behavior

Even hitting cancel, the applications are started. While you never get access to the user interface itself, this is still a large issue. In some cases, the automated behavior of the apps are initiated (ex. podbird will start downloading new podcasts) and depending on what the user has on the dock, this could be any code running with full network access.

### Logfiles and additional information

I'm not sure what log files are helpful here.

**Comment by UniversalSuperBox** [comment]
Hi @stuarttempleton,

Being able to start apps from the lock screen is by design on the Unity8 greeter. However, it was understood that this could be a threat for some users. For that reason, you can turn this off by going to Settings -> Security & Privacy -> Locking and unlocking, then unchecking "When locked, allow: Launcher".

Does this resolve the issue?

**Comment by stuarttempleton** [comment]
Thanks for the clarification!

Using that setting does prevent the apps from being started on the lock screen, which turns the insecure feature off, but I don't believe it addresses the security issue itself.

It brings up some concerns about the design, though. This isn't a standard user experience on devices like this, so it really needs to be explained to the user if it's intended to be default. I don't think it's reasonable to expect new users to understand that paradigm by default, which sets them up for a potentially insecure experience out of the box. 

Depending on the use cases that design is intended to solve (quick access to camera, reply to notification, answer call, etc...) a new design might be warranted, but until a security-focused design is in place, I suggest that anything that is user-initiated *before* authentication be subject to security review of some kind (community is fine).  For example, the community and the Ubuntu Touch crew can (and probably do) review the camera app and dialer app with little issue, so stuff like that could make sense, but a random side-loaded app can't be assumed to meet any reasonable security standards.

I'm realizing that perhaps this is part of a larger conversation that needs to happen in some other context or discussion channel? 

Either way: My opinion is that this design is not optimal from an infosec stand point. Low hanging fruit, maybe disable it by default and make the user enable it, indicating that they might understand and want the feature. Possibly, take the opportunity to revisit the language to ensure it communicates the full scope of what is happening.

**Comment by dobey** [comment]
How is this a security issue exactly? At what point is the security of the device actually compromised through this design?

One cannot install an app on the device from the lock screen. One cannot see data in said apps from the lock screen. One must first unlock the device to access anything. Heck, the app will be started, but in the paused state (it can't actually process anything), due to the app life cycle management in Ubuntu Touch.

**Comment by stuarttempleton** [comment]
It's a security issue because it is a situation where an unauthenticated user can trigger the launching of application and execution of code and it's both the default behavior and not part of an expected user experience paradigm.

I think you might be right about the lifecycle, though. My example was podbird and I suspect that I may have used UT Tweak Tool to prevent that app (and some others) from "backgrounding", so it might have just gone about its business. I think this illustrates part of the issue, though. because I didn't expect this paradigm, I didn't understand that not backgrounding an app would leave me open to that kind of thing.  

As Dalton kindly added, it is acknowledged as a threat for some users, and this is an example of that use case. I'm adding that it's more a more common case and probably needs to be re-considered as a default, especially as UT starts to hit privacy focused devices and communities.

**Comment by stuarttempleton** [comment]
The more I think about it, this is a known issue and maybe a security-adjacent feature request related to default behavior of the greeter. What do you guys think?

**Comment by dobey** [comment]
> It's a security issue because it is a situation where an unauthenticated user can trigger the launching of application and execution of code and it's both the default behavior and not part of an expected user experience paradigm.

One cannot cause the system to execute _arbiltrary_ code in this manner. They can only cause the system to launch an app which is pinned to the launcher. The app is then immediately paused, and there is no way for the person to actually use the app, because the screen is not magically unlocked. This is NOT a security issue.

> As Dalton kindly added, it is acknowledged as a threat for some users, and this is an example of that use case. I'm adding that it's more a more common case and probably needs to be re-considered as a default, especially as UT starts to hit privacy focused devices and communities.

Well, it is not a threat in the sense that starting an app will somehow result in a breach of the system or data being stolen. The only argument one can make here is that it is a _privacy_ issue (not a security one), because one can see some of the apps the user may use most, are currently running, etc… Or they may be able to unpin them from launcher (I've not actually tested that). But again, there is no security issue, as there is no way to breach the lock screen or access data through this action.

I am not opposed to the options to block launcher and indicators panel access on lock screen being enabled by default, though it will make for more confusing UX by default. So, I would suggest instead to come up with a plan that satisfies your concerns, but also doesn't make the UX more difficult for the general populace, by default.

**Comment by stuarttempleton** [comment]
I didn't say anything about the lock screen being "magically" unlocked. This conversation is neither productive of friendly. 

Process this information however best benefits the project. Have a good week folks! I'm tapping out.

**Comment by Flohack74** [comment]
Well you have to see if you press the "security issue" button then everything you are claiming will be reviewed very precisely by us and you have to expect contradiction and a lot of questions. Such conversations are not friendly most of the time, especially if it starts to being opinion-based if this is a security hole or not.

I suggest that if you can actually proof with a demo that this is a security hole and you can exploit anything we might take it further.

**Comment by stuarttempleton** [comment]
I don't mind that we disagree on the nature of this issue at all. In fact, I actually see some of Rodney's points. It's a known, by-design behavior and the issue is closed. I see no reason to take it further unless you do. 

If someone in your community should expect unfriendly conversations from members of UBPorts just for bringing up a potential security issue (even if it isn't one), then your community has some serious toxicity issues that stem directly from the organization. If you fee the need to defend "not friendly" behavior on a closed issue, you need to get your house in order because It's probably not an isolated situation. 

Anyway, like I said: Process this information however best benefits the project. Have a good week folks! I'm tapping out.

**Comment by dobey** [comment]
> Such conversations are not friendly most of the time, especially if it starts to being opinion-based if this is a security hole or not.

I disagree. There was nothing unfriendly in this issue thread. And there's no reason for anything unfriendly to be expected from a discussion about a potential security issue. But simply declaring something a security issue as a means to get change is not acceptable. This is by definition not a security issue, and would not be accepted as a CVE. Otherwise Android and iOS would already have had the same thing lodged against them very loudly as well.

To be a security issue, something must actually breach security so that you have escalated privileges, access to otherwise inaccessible data, etc… Nothing stated in this ticket meets the criteria for a security issue. Declaring the conversation unfriendly simply because you disagree and don't accept the multiple attempts to explain it to you, is not a good way to move forward with a proposed change.

**Comment by Flohack74** [comment]
Yeah me too I dont see any unfriendlyness so far.
